backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java

```diff
@@ -0,0 +1,245 @@
+                                  + " VALUES (:repository_id, :commit, :versioning_key, :entityId, :entity_type, :blob_hash, :config_seq_number)";
+                          handle
+                              .createUpdate(blobMappingQueryStr)
+                              .bind("repository_id", versioningEntry.getRepositoryId())
+                              .bind("commit", versioningEntry.getCommit())
+                              .bind("versioning_key", locationEntry.getKey())
+                              .bind("entityId", entityId)
+                              .bind("entity_type", entity_type)
+                              .bind("blob_hash", rs.getString("blob_hash"))
+                              .bind("config_seq_number", rs.getString("config_seq_number"))
+                              .execute();
+
+                          // Insert prepare hyperparameter elemMappings
+                          if (rs.getInt("hyperparameter_type") == HYPERPARAMETER) {
+                            String hemeStr =
+                                "INSERT INTO hyperparameter_element_mapping (name, int_value, float_value, string_value, entity_type, "
+                                    + entityIdReferenceColumn
+                                    + " ) VALUES (:name, :int_value, :float_value, :string_value, :entity_type, :entity_id )";
+                            handle
+                                .createUpdate(hemeStr)
+                                .bind("name", rs.getString("name"))
+                                .bind(
+                                    "int_value",
+                                    rs.getInt("int_value") != 0 ? rs.getInt("int_value") : null)
+                                .bind("float_value", rs.getFloat("float_value"))
+                                .bind("string_value", rs.getString("string_value"))
+                                .bind("entity_type", entity_type)
+                                .bind("entity_id", entityId)
+                                .execute();
+                          }
+                          return rs;
+                        })
+                    .list();
+              }
+            }
+          }
+        });
+  }
+
+  /**
+   * @param versioningEntry : versioningEntry
+   * @return returns a map from location to an Entry of BlobExpanded and sha
+   * @throws ModelDBException ModelDBException
+   */
+  private InternalFuture<Map<String, Map.Entry<BlobExpanded, String>>> validateVersioningEntity(
+      VersioningEntry versioningEntry) {
+    String errorMessage = null;
+    if (versioningEntry.getRepositoryId() == 0L) {
+      errorMessage = "Repository Id not found in VersioningEntry";
+    } else if (versioningEntry.getCommit().isEmpty()) {
+      errorMessage = "Commit hash not found in VersioningEntry";
+    }
+
+    if (errorMessage != null) {
+      throw new ModelDBException(errorMessage, io.grpc.Status.Code.INVALID_ARGUMENT);
```

Comment:
Don't throw exceptions. You're returning a future, so just return a failed future. Throwing exceptions is now an anti-pattern as it breaks the execution chain. It's only acceptable if you're not returning a future.

Reply:
Fixed